### PR TITLE
add long.__float__

### PR DESCRIPTION
**Pull request by aisk**
https://github.com/dropbox/pyston/issues/450

### CODE REVIEW

**Comment by kmod** [comment]
Hi, can we get a quick test for this?  I know it's a bit weird to ask for since the motivation of this was to get the CPython test to work, but until we have that test enabled, the code you added is untested.  Just something simple like calling `print float(1L), (2L).__float__()`.

**Comment by undingen** [comment]
And you have to change the ```float``` to a ```double``` otherwise ```(16777217L).__float__()``` will return ```16777216.0``` :-D (haven't tried it but the code looks like it)

File: src/runtime/long.cpp [hunk 1]
```diff
@@ -688,6 +688,18 @@ Box* longInt(Box* v) {
         return new BoxedInt(n);
 }
 
+Box* longFloat(BoxedLong* v) {
+    if (!isSubclass(v->cls, long_cls))
+        raiseExcHelper(TypeError, "descriptor '__float__' requires a 'long' object but received a '%s'",
+                       getTypeName(v));
+    double inf = std::numeric_limits<double>::infinity();
+    double f = mpz_get_d(v->n);
+    if (f == inf || f == -inf) {
```
Comment (kmod) [inline src/runtime/long.cpp]: Can we switch this to `std::isinf(f)`?

**Comment by kmod** [comment]
So looking at the CPython code for this, it looks like their `__float__` implementation is just a wrapper around `PyLong_AsDouble()`.  I didn't think about it when you started the PR, but it seems like we should do that too in order to reduce duplication.  Case in point: PyLong_AsDouble should do the same overflow checking that you added here :)

Also, minor thing, but my preference is to keep trivial merges out of PRs.  Would you mind rebasing at some point?

**Comment by aisk** [comment]
Got, I will do this :)

**Comment by kmod** [comment]
Superseded by #683

**Pull request closed without merging**